stravalib/model.py

```diff
@@ -116,6 +125,80 @@ def __repr__(self):
         )
 
 
+class DeprecatedSerializableMixin(BaseModel):
+    """
+    Provides backward compatibility with legacy BaseEntity
+    """
+
+    @classmethod
+    def deserialize(cls, v):
+        """
+        Creates and returns a new object based on serialized (dict) struct.
+        """
+        warn_method_deprecation(
+            cls,
+            "deserialize()",
+            "parse_obj()",
+            "https://docs.pydantic.dev/usage/models/#helper-functions",
+        )
+        return cls.parse_obj(v)
+
+    def from_dict(self, v):
+        """
+        Deserializes v into self, resetting and ond/or overwriting existing fields
+        """
+        warn_method_deprecation(
+            self.__class__.__name__,
+            "from_dict()",
+            "parse_obj()",
+            "https://docs.pydantic.dev/usage/models/#helper-functions",
+        )
+        # Ugly hack is necessary because parse_obj does not behave in-place but returns a new object
+        self.__init__(**self.parse_obj(v).dict())
+
+    def to_dict(self):
+        """
+        Returns a dict representation of self
+        """
+        warn_method_deprecation(
+            self.__class__.__name__,
+            "to_dict()",
+            "dict()",
+            "https://docs.pydantic.dev/usage/exporting_models/",
+        )
+        return self.dict()
+
+
+class BackwardCompatibilityMixin:
+    """
+    Mixin that intercepts attribute lookup and raises warnings or modifies return values
+    based on what is defined in the following class attributes:
+    * _deprecated_fields (TODO)
+    * _unsupported_fields (TODO)
+    * _field_conversions
+    * _unit_registry
+    """
+
+    def __getattribute__(self, attr):
+        value = object.__getattribute__(self, attr)
+        if attr in ["_field_conversions", "_unit_registry"]:
+            return value
+        try:
+            if attr in self._field_conversions:
+                value = self._field_conversions[attr](value)
+        except AttributeError:
+            # Current model class has no field conversions defined
+            pass
+        try:
+            if attr in self._unit_registry:
+                # Return a Quantity
+                value = UnitConverter(self._unit_registry[attr])(value)
```

Comment:
`_unit_registry` is a class attribute that maps field names to units and uses the unit converter to transforms these fields to unit-full quantities when requested